src/sdk/wire_bytes.rs

```diff
@@ -27,41 +28,62 @@ pub fn encode<T: Serialize>(payload: &T) -> TofnResult<BytesVec> {
     })
 }
 
+/// Serialize a value using bincode and log errors
 pub fn serialize<T: ?Sized>(value: &T) -> TofnResult<BytesVec>
 where
     T: serde::Serialize,
 {
-    match bincode::serialize(value) {
-        Ok(bytes) => Ok(bytes),
-        Err(err) => {
-            error!("serialization failure: {}", err.to_string());
-            Err(TofnFatal)
-        }
-    }
+    // Create serialization options for bincode.
+    // The default options don't bound pre-allocation size,
+    // use little-endian and varint encoding, and reject trailing bytes.
+    let options = DefaultOptions::new()
+        .with_no_limit()
```

Comment:
@ggutoski I would like to use the `Bounded` bincode config option, but is there a nicer way than adding const generics on all these functions? Or is this option not too concerning now that we enforce a limit in the sdk instead?

Reply:
Quoting https://github.com/axelarnetwork/tofn/issues/169#issuecomment-931689985 :

>  I had too much trouble trying to use bincode to do this so we'll just do it the old-fashioned way---with an `if` statement before calling `bincode::deserialize`

If you can figure out how to make `Bounded` work then go for it.  Is there a reason to prefer `Bounded` over what we currently have?

I agree that the const generics add a lot of cruft.  It's yet another item on the growing list of type parameters `<F, K, P, const MAX_MSG_IN_LEN: usize>` common to everything in the sdk.  I'm open to suggestion.  If you find a good solution then perhaps we could also use it to replace this:

https://github.com/axelarnetwork/tofn/blob/1809a934907884dd4d0393f9dba71e7b58da1909/src/sdk/api.rs#L14-L16

Reply:
Haven't thought this through at all, but we could declare a trait
```rust
trait RoundParams {
  type FinalOutput;
  type ShareId;
  type PartyId;
  const MAX_MSG_IN_LEN: usize;
  // etc
}
```
and then replace all instances of `<F, K, P, const MAX_MSG_IN_LEN: usize>` with `<P: RoundParams>` and then refer to `P::FinalOutput` instead of `F`, etc.  That's nice because we can change the type parameters at only one place and reduce boilerplate.

Reply:
I missed your comment on the issue :/ I think the `if` check we have is pretty good, but I was thinking of using `Bounded` as a safety check for anyone using the `encode/decode` functions in general. But I don't think it's super important. And yeah, the types for the bincode options are not intuitive, but with what I'm doing here, it could be achieved if we wanted it.
I also really like the `trait RoundParams` idea.

Reply:
>  I think the `if` check we have is pretty good, but I was thinking of using `Bounded` as a safety check for anyone using the `encode/decode` functions in general. But I don't think it's super important.

That's not a bad idea.  There's nothing wrong with a redundant line of defence.  We could set the limit much higher than the current check in `Round::msg_in`.  eg 1MB?  The limit must be high enough to comfortably fit anything we might encode including non-constant-size data blobs such as key recovery data.

Seems like you've already done most of the work to use `Bounded`, in which case it's trivial to tweak what you already have with a suitable limit.  But if it's too much additional work then we can do without.  I'll leave it up to you.